### DISCUSSION TITLE
Issues triggering an alert

### DISCUSSION THREAD

**Discussion by juaneloxx** [jertel/elastalert2#428] [Q&A] [answered]
Hello, im new to elastalert, i have a fresh install and just added 1 alert for testing
I'm still not sure if the issue is my rule or is something else.
When I test the rule in debug mode with `elastalert --config /opt/elastalert2/config.yaml --rule alerta_conexion_rdp_ad.yml
`
I get this result. https://pastebin.com/9y6a153d

But when i try with `elastalert --config /opt/elastalert2/config.yaml --rule alerta_conexion_rdp_ad.yml`
which is supposed to trigger the alert and actually send the alert i get this. https://pastebin.com/NFftB5m7

as you can see my conf file is on /opt/elastalert2/config.yaml
This is my config file.
```
rules_folder: /opt/elastalert2/rules/
run_every:
  minutes: 1
buffer_time:
  minutes: 15
es_host: localhost
es_port: 9200
writeback_index: elastalert_status
writeback_alias: elastalert_alerts
alert_time_limit:
 days: 2
es_username: *********
es_password: ********* 
```

And just in case, this is the rule im testing.

```
name: Acceso AD RDP
description: test juanelo
type: any
index: winlogbeat*
run_every:
 minutes: 15
timeframe:
 hours: 24
filter:
  - query:
      query_string:
        query: 'winlog.channel:"Security" AND winlog.event_data.AuthenticationPackageName:"Negotiate" AND winlog.event_id:"4624" AND winlog.event_data.LogonType:"10" AND winlog.event_data.TargetUserName:Admin*'

include: ["user.name", "source.ip", "event.action", "event.outcome"]

alert:
 - telegram:
       telegram_bot_token:*************
       telegram_room_id:*************
```

I want to receive the alerts on a telegram group chat.

sorry about my english, not my native language.

**Comment by jertel** [comment, marked as answer]
It's difficult to read your post. Use triple back ticks to denote a multi line block of code instead of using single back ticks on each line.

It looks like you have a colon after the telegram alerter list item which probably shouldn't be there.

**Reply by juaneloxx** [reply]
which colon shouldnt be there? 
`- telegram:`<-that one?
BTW thanks about the triple backticks

**Reply by jertel** [reply]
Yes that's the one. 

The telegram_* keys should not be indented.

**Reply by juaneloxx** [reply]
So, i think the problem with the rule should be fixed?
I removed the indent and had to put the room_id between ' '
Now when i run the command the rule loads but i get this err now https://pastebin.com/M0QkkKpf
The error is this repeating over and over
```
ERROR:elastalert:Error finding recent pending alerts: NotFoundError(404, 'index_not_found_exception', 'no suc                                                                                                                                h index [elastalert_status]', elastalert_status, index_or_alias) {'query': {'bool': {'must': {'query_string':                                                                                                                                 {'query': '!_exists_:aggregate_id AND alert_sent:false'}}, 'filter': {'range': {'alert_time': {'from': '2021                                                                                                                                -08-25T02:15:16.341989Z', 'to': '2021-08-27T02:15:16.342022Z'}}}}}, 'sort': {'alert_time': {'order': 'asc'}}}
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/elastalert2-2.2.1-py3.6.egg/elastalert/elastalert.py", line 17                                                                                                                                27, in find_recent_pending_alerts
    res = self.writeback_es.search(index=self.writeback_index, body=query, size=1000)
  File "/usr/local/lib/python3.6/site-packages/elasticsearch-7.0.0-py3.6.egg/elasticsearch/client/utils.py",                                                                                                                                 line 84, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/elasticsearch-7.0.0-py3.6.egg/elasticsearch/client/__init__.py                                                                                                                                ", line 811, in search
    "GET", _make_path(index, "_search"), params=params, body=body
  File "/usr/local/lib/python3.6/site-packages/elasticsearch-7.0.0-py3.6.egg/elasticsearch/transport.py", lin                                                                                                                                e 318, in perform_request
    status, headers_response, data = connection.perform_request(method, url, params, body, headers=headers, i                                                                                                                                gnore=ignore, timeout=timeout)
  File "/usr/local/lib/python3.6/site-packages/elasticsearch-7.0.0-py3.6.egg/elasticsearch/connection/http_re                                                                                                                                quests.py", line 91, in perform_request
    self._raise_error(response.status_code, raw_data)
  File "/usr/local/lib/python3.6/site-packages/elasticsearch-7.0.0-py3.6.egg/elasticsearch/connection/base.py                                                                                                                                ", line 131, in _raise_error
    raise HTTP_EXCEPTIONS.get(status_code, TransportError)(status_code, error_message, additional_info)
elasticsearch.exceptions.NotFoundError: NotFoundError(404, 'index_not_found_exception', 'no such index [elast                                                                                                                                alert_status]', elastalert_status, index_or_alias)
WARNING:elasticsearch:GET http://localhost:9200/elastalert_status/_search?size=1000 [status:404 request:0.006                                                                                                                                s]
```

**Reply by nsano-rururu** [reply]
@juaneloxx 

Are you reading the documentation?

https://elastalert2.readthedocs.io/en/latest/ruletypes.html#telegram

```yaml
alert:
  - "telegram"
telegram_bot_token: "bot_token"
telegram_room_id: "chat_id"
```

**Reply by nsano-rururu** [reply]
https://gist.github.com/soulsearcher/30d3c2ea3c89e53b01e87b83b8ebe91b

**Reply by juaneloxx** [reply]
Yes, ive been reading, since my native language is not english, im not familiar with some terms that are in the documentation and i've tried the way you just showed me but i get the same exact error.
currently the rule looks like this:
```
name: Acceso AD RDP
description: test juanelo
type: any
index: winlogbeat*
run_every:
 minutes: 15
timeframe:
 hours: 24
filter:
  - query:
      query_string:
        query: 'winlog.channel:"Security" AND winlog.event_data.AuthenticationPackageName:"Negotiate" AND winlog.event_id:"4624" AND winlog.event_data.LogonType:"10" AND winlog.event_data.TargetUserName:Admin*'

include: ["user.name", "source.ip", "event.action", "event.outcome"]

alert:
- "telegram"
telegram_bot_token: "**********:*********************************_**"
telegram_room_id: "*****************"
```
Ive tried without the double quotes on token and single quote on room id
and like its shown above with double quotes on both token and id

**Reply by nsano-rururu** [reply]
My mother tongue is not English either.

**Reply by nsano-rururu** [reply]
I get'index_not_found_exception' in the error message. Does the specified index "winlogbeat *" exist in Elasticsearch?

**Reply by juaneloxx** [reply]
it does
![image](https://user-images.githubusercontent.com/89549332/131069248-a52b74de-ece6-400e-a32b-e04924bf4731.png)

**Reply by juaneloxx** [reply]
I'm still new to this but i assume winlogbeat* includes all the indices shown below
![image](https://user-images.githubusercontent.com/89549332/131069405-735f6fcc-29ca-4aac-8f57-5ff28203439e.png)

**Reply by nsano-rururu** [reply]
Are you running elastalert-create-index?
https://elastalert2.readthedocs.io/en/latest/running_elastalert.html#setting-up-elasticsearch

**Reply by nsano-rururu** [reply]
Try changing from `index: winlogbeat*'` to `index: winlogbeat-*`.
If that doesn't work, delete the index starting with elastalert on the kibana screen. Run elastalert-create-index again. The reason for asking for the latter work is that my guess is that the index of elastalert is currently inaccessible from elastalert2, so I think that deleting the index and recreating it will solve the problem.

**Reply by nsano-rururu** [reply]
That's all I can answer. It is up to you to decide whether to try it or solve it yourself.

**Reply by juaneloxx** [reply]
That actually worked thx a lot :)
